rs/registry/transport/src/lib.rs

```diff
@@ -113,6 +113,38 @@ impl From<Error> for RegistryError {
     }
 }
 
+impl registry_mutation::Type {
+    pub fn is_delete(self) -> bool {
+        // Do not simply replace this with self == Delete, because that assumes
+        // the reader knows that all other mutation types are not some other
+        // flavor of deletion. Whereas, this match proves (in writing) that we
+        // really individually considered every single mutation types.
+        match self {
+            registry_mutation::Type::Delete => true,
+
+            registry_mutation::Type::Insert
+            | registry_mutation::Type::Update
+            | registry_mutation::Type::Upsert => false,
+        }
+    }
+
+    /// Returns whether key being touched
+    ///
+    ///   * must be present. In this case, Some(true) is returned.
+    ///   * must be absence. In this case, Some(false) is returned.
+    ///   * can be present or absent: In this case None is returned.
+    pub fn requires_already_present(self) -> Option<bool> {
+        match self {
+            registry_mutation::Type::Upsert => None, // Anything goes!
+
+            // Destructive operations must have SOME value to destroy.
+            registry_mutation::Type::Delete | registry_mutation::Type::Update => Some(true),
+
+            registry_mutation::Type::Insert => Some(false), // Do NOT clobber!
```

Comment:
This is a surprising semantic based on the name "requires_already_present" which would indicate a boolean return.  I wouldn't expect Option<bool> to give me 3 options based on that name.  Maybe a simple enum that says "MustBePresent" "MustNotBePresent", "MayBePresent" would be clearer?